File: swiftUI/Data/Repositories/PhucTvRepository.swift
import Foundation

protocol PhucTvRepository: Sendable {
    func loadHome() async throws -> [PhucTvHomeSection]
    func loadNavbar() async throws -> [PhucTvNavbarItem]
    func loadDetail(slug: String) async throws -> PhucTvMovieDetail
    func loadPreview(slug: String) async throws -> PhucTvMovieDetail
    func loadSearchFilters() async throws -> PhucTvSearchFilterData
    func loadSearchResults(
        categoryId: Int?,
        countryId: Int?,
        typeRaw: String,
        year: String,
        orderBy: String,
        isChieuRap: Bool,
        is4k: Bool,
        search: String,
        pageNumber: Int
    ) async throws -> PhucTvSearchResults
    func loadEpisodeSources(
        movieID: Int,
        episodeID: Int,
        server: Int
    ) async throws -> [PhucTvPlaySource]
    func loadPopupAd() async throws -> PhucTvPopupAdConfig?
}

File: android-compose/app/src/main/java/com/motchill/androidcompose/data/repository/PhucTVRepository.kt
package com.motchill.androidcompose.data.repository

import com.motchill.androidcompose.domain.model.HomeSection
import com.motchill.androidcompose.domain.model.MovieDetail
import com.motchill.androidcompose.domain.model.NavbarItem
import com.motchill.androidcompose.domain.model.PlaySource
import com.motchill.androidcompose.domain.model.PopupAdConfig
import com.motchill.androidcompose.domain.model.SearchFilterData
import com.motchill.androidcompose.domain.model.SearchResults

interface PhucTVRepository {
    suspend fun loadHome(): List<HomeSection>
    suspend fun loadNavbar(): List<NavbarItem>
    suspend fun loadDetail(slug: String): MovieDetail
    suspend fun loadPreview(slug: String): MovieDetail
    suspend fun loadSearchFilters(): SearchFilterData
    suspend fun loadSearchResults(
        categoryId: Int? = null,
        countryId: Int? = null,
        typeRaw: String = "",
        year: String = "",
        orderBy: String = "UpdateOn",
        isChieuRap: Boolean = false,
        is4k: Boolean = false,
        search: String = "",
        pageNumber: Int = 1,
    ): SearchResults

    suspend fun loadEpisodeSources(
        movieId: Int,
        episodeId: Int,
        server: Int = 0,
    ): List<PlaySource>

    suspend fun loadPopupAd(): PopupAdConfig?
}

